storage/scylla_listener.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+#include <zmq.h>
+#include "accelerator/config.h"
+#include "cclient/service.h"
+#include "scylla_api.h"
+
+#define SCYLLA_LISTENER_LOGGER "SCYLLA_LISTENER"
+
+static logger_id_t logger_id;
+void scylla_listener_logger_init() { logger_id = logger_helper_enable(SCYLLA_LISTENER_LOGGER, LOGGER_DEBUG, true); }
+
+int scylla_listener_logger_release() {
+  logger_helper_release(logger_id);
+  if (logger_helper_destroy() != RC_OK) {
+    log_critical(logger_id, "%s.\n", SCYLLA_LISTENER_LOGGER);
+    return EXIT_FAILURE;
+  }
+  return 0;
+}
+
+//  Receive ZMQ string from socket and convert into C string
+//  Caller must free returned string. Returns NULL if the context
+//  is being terminated.
+static char* receive_zmq_string_to_heap(void* socket) {
+#define BUFFER_LEN 450
```

Comment:
Move macro `BUFFER_LEN` out of this function body. That is, before this function.